### PR TITLE
linux-yocto: enable various remotprocs for qcm6490 boards

### PR DESCRIPTION
**Pull request by adhudase**
Add support for the ADSP, CDSP and WPSS remoteprocs found on SC7280 and enable various applicable remoteproc nodes for qcm6490-idp and qcs6490-rb3gen2.

This has dependency on PR#587.

### CODE REVIEW

File: recipes-kernel/linux/linux-yocto/qcm6490-board-dts/0001-FROMLIST-arm64-dts-qcom-qcm6490-idp-Enable-various-r.patch [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
++	status = "okay";
++};
++
++&remoteproc_mpss {
++	firmware-name = "qcom/qcm6490/modem.mdt";
++	status = "okay";
++};
++
++&remoteproc_wpss {
++	firmware-name = "qcom/qcm6490/wpss.mdt";
```
Comment (lumag) [inline recipes-kernel/linux/linux-yocto/qcm6490-board-dts/0001-FROMLIST-arm64-dts-qcom-qcm6490-idp-Enable-various-r.patch:40]: mbn only

File: recipes-kernel/linux/linux-yocto/qcm6490-board-dts/0002-FROMLIST-arm64-dts-qcom-qcs6490-rb3gen2-Enable-vario.patch [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
++	status = "okay";
++};
++
++&remoteproc_cdsp {
++	firmware-name = "qcom/qcm6490/cdsp.mbn";
++	status = "okay";
++};
++
++&remoteproc_wpss {
++	firmware-name = "qcom/qcm6490/wpss.mdt";
```
Comment (lumag) [inline recipes-kernel/linux/linux-yocto/qcm6490-board-dts/0002-FROMLIST-arm64-dts-qcom-qcs6490-rb3gen2-Enable-vario.patch:35]: mbn

**Review by lumag — CHANGES_REQUESTED** [review]

File: recipes-kernel/linux/linux-yocto/qcm6490-board-dts/0001-FROMLIST-arm64-dts-qcom-qcm6490-idp-Enable-various-r.patch (hunk 1 above)
Comment (adhudase) [reply recipes-kernel/linux/linux-yocto/qcm6490-board-dts/0001-FROMLIST-arm64-dts-qcom-qcm6490-idp-Enable-various-r.patch:40]: Ack. This was intentional for draft PR as I don't see mbn format for modem and wpss yet. I'll update it to mbn when ready.

File: recipes-kernel/linux/linux-yocto/qcm6490-board-dts/0002-FROMLIST-arm64-dts-qcom-qcs6490-rb3gen2-Enable-vario.patch (hunk 2 above)
Comment (adhudase) [reply recipes-kernel/linux/linux-yocto/qcm6490-board-dts/0002-FROMLIST-arm64-dts-qcom-qcs6490-rb3gen2-Enable-vario.patch:35]: Ack. This was intentional for draft PR as I don't see mbn format for modem and wpss yet. I'll update it to mbn when ready.

Comment (lumag) [reply recipes-kernel/linux/linux-yocto/qcm6490-board-dts/0002-FROMLIST-arm64-dts-qcom-qcs6490-rb3gen2-Enable-vario.patch:35]: Use https://github.com/linux-msm/pil-squasher

**Comment by adhudase** [comment]
Sure, fixing the rebase. I'll update this PR once upstream patches are updated by Komal.

**Comment by lumag** [comment]
@adhudase But there is one already, it was picked up by Bjorn: https://lore.kernel.org/linux-arm-msm/171373856754.1196479.8348340567259271476.b4-ty@kernel.org/

**Comment by lumag** [comment]
@adhudase I mean, there is an updated series by Komal, it has been picked up by Bjorn.

**Comment by adhudase** [comment]
Hi @lumag, there was internal discussion around your comments regarding separate firmware folders for qcs6490 vs qcm6490 and modem vs modem-nm, which is not yet concluded. I will wait for that to conclude and then update this PR accordingly.

Thanks

**Comment by lumag** [comment]
@adhudase the patches are already picked up for -next. So if there are any changes, you have to base it on top of the accepted patches.

**Comment by lumag** [comment]
@adhudase any updates? A cadence of several weeks is really depressing

**Comment by lumag** [comment]
@adhudase any updates

**Comment by adhudase** [comment]
Hi @lumag, sorry for delay in response; somehow missed your message. I still don't see zap shader in no-login release and modem is not expected to be shipped in no-login. So if you agree can we go ahead with firmware recipe without these?
Thanks

**Comment by adhudase** [comment]
Seems mainline went ahead with mdt for modem on qcs6490;missed mbn. I've modified that to mbn here.
qcom/qcs6490/modem.mdt -> qcom/qcs6490/modem.mbn

**Comment by lumag** [comment]
@adhudase thanks! please mark it as 'ready', I'll merge it. Also please send a patch to linux-arm-msm, changing `.mdt` to `.mbn`. Thank you!

**Review by lumag — APPROVED** [review]

**Comment by adhudase** [comment]
Hi @lumag, can we merge this PR?

**Comment by lumag** [comment]
For some reason, I didn't get the notification that it's marked as ready. Merged now.

**Comment by Thejesh99** [comment]
i have a qualcomm robotics RB3 gen2 vision kit 
can u please provide me a yocto guide for qualcomm robotics RB3 gen2 vision kit 
or any github repo and distro file
Regards
Thejesh

**Pull request merged**